File: solutions/missing-number.cpp
/*
 *Solution for problem 268
 *https://leetcode.com/problems/missing-number/
 */
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum = 0, size = nums.size();
        for(int n: nums){
            sum += n;
        }
        int totalSum = ((size+1) * (size))/2;
        return totalSum - sum;
    }
};

File: solutions/missing-number.py
"""
268. Missing Number
- https://leetcode.com/problems/missing-number/
- https://leetcode.com/explore/featured/card/march-leetcoding-challenge-2021/588/week-1-march-1st-march-7th/3659/
"""

# Method 1
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        numsLen = len(nums)
        totalSum = numsLen * (numsLen + 1) // 2
        return totalSum - sum(nums)



